akka-actor/src/main/scala/akka/dispatch/BatchingExecutor.scala

```diff
@@ -89,21 +89,22 @@ private[akka] trait BatchingExecutor extends Executor {
         parentBlockContext = prevBlockContext
         try processBatch(this) catch {
           case t: Throwable ⇒
+            parentBlockContext = null
             resubmitUnbatched()
             throw t
-        } finally {
-          _tasksLocal.remove()
-          parentBlockContext = null
-        }
+        } finally _tasksLocal.remove()
+        parentBlockContext = null
       }
     }
 
     override def blockOn[T](thunk: ⇒ T)(implicit permission: CanAwait): T = {
+      // need to rescue the parent BC before re-running this Batch
+      val parent = parentBlockContext
```

Comment:
Nope, that line would not help any: either we have a bug that needs fixing, and then an NPE would tell the same story, or we don’t have a bug.